pkg/controller/route/resources/virtual_service.go

```diff
@@ -98,13 +110,23 @@ func makeVirtualServiceSpec(u *v1alpha1.Route, targets map[string][]traffic.Revi
 }
 
 func getRouteDomains(targetName string, u *v1alpha1.Route, domain string) []string {
+	var domains []string
 	if targetName == "" {
-		// Nameless traffic targets correspond to two domains: the Route.Status.Domain, and also the FQDN
-		// of the Route's headless Service.
-		return []string{domain, names.K8sServiceFullname(u)}
+		// Nameless traffic targets correspond to many domains: the
+		// Route.Status.Domain, and also various names of the Route's
+		// headless Service.
+		domains = []string{domain,
+			names.K8sServiceFullname(u),
+			fmt.Sprintf("%s.%s.svc", u.Name, u.Namespace),
+			fmt.Sprintf("%s.%s", u.Name, u.Namespace),
+		}
+		if u.Namespace == "default" {
```

Comment:
I'm not sure this is how it works, is it?

Can you link to documentation stating this behavior?

Reply:
This wasn't right... fixed.  For a Pod in the same namespace only the service name is needed.  So we should allow `Host:<route>` regardless what namespace it is in.

Doc is here
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods